kivy/uix/carousel.py

```diff
@@ -284,6 +274,8 @@ def _next_slide(self):
     _next = ObjectProperty(None, allownone=True)
     _offset = NumericProperty(0)
     _touch = ObjectProperty(None, allownone=True)
+    _prev_equals_next = BooleanProperty(False)
+    _prioritize_next = BooleanProperty(False)
```

Comment:
Unless the properties are needed for the users to be able to bind to, or it really simplifies the code, it is better to use a `@property` or normal python property rather than a KivyProperty.

That's because with Kivy properties it is harder to trace the code logic (due to action at a distance), and they add loading performance cost to the widgets.

So for example, `_prev_equals_next` doesn't seem to benefit at all from being a KivyProperty, so it should just be a normal property. But even `_prioritize_next`, I would make it a normal property and inline the code in `on__prioritize_next` at the one place you set it. Or you can make it a python `@property` with a setter that does the stuff currently in `on__prioritize_next` when set.

Reply:
I agree with you. I'll make those into non KivyProperty.